diff --git a/ui/views/examples/BUILD.gn b/ui/views/examples/BUILD.gn
index ca5edf49f4727..949d8486e8020 100644
--- a/ui/views/examples/BUILD.gn
+++ b/ui/views/examples/BUILD.gn
@@ -6,6 +6,7 @@ import("//build/config/ui.gni")
 import("//testing/test.gni")
 import("//tools/grit/grit_rule.gni")
 import("//ui/gl/features.gni")
+import("//third_party/icu/config.gni")

 if (is_mac) {
   import("//build/apple/tweak_info_plist.gni")
@@ -371,7 +372,6 @@ if (is_mac) {
     testonly = true

     sources = [
-      "$root_out_dir/icudtl.dat",
       "$root_out_dir/ui_test.pak",
       "$root_out_dir/views_examples_resources.pak",
     ]
@@ -379,8 +379,10 @@ if (is_mac) {
       ":views_examples_resources_pak",
       "//ui/resources:ui_test_pak",
     ]
-    deps = [ "//third_party/icu:icudata" ]
-
+    if (icu_use_data_file) {
+      sources += [ "$root_out_dir/icudtl.dat" ]
+      deps += [ "//third_party/icu:icudata" ]
+    }
     outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
   }

