diff --git a/ui/views/examples/BUILD.gn b/ui/views/examples/BUILD.gn
index 500dd0d66f6c7..fcd5e836ba295 100644
--- a/ui/views/examples/BUILD.gn
+++ b/ui/views/examples/BUILD.gn
@@ -5,6 +5,7 @@
 import("//build/config/ui.gni")
 import("//testing/test.gni")
 import("//tools/grit/grit_rule.gni")
+import("//third_party/icu/config.gni")
 
 if (is_mac) {
   import("//build/apple/tweak_info_plist.gni")
@@ -366,7 +367,6 @@ if (is_mac) {
     testonly = true
 
     sources = [
-      "$root_out_dir/icudtl.dat",
       "$root_out_dir/ui_test.pak",
       "$root_out_dir/views_examples_resources.pak",
     ]
@@ -374,7 +374,10 @@ if (is_mac) {
       ":views_examples_resources_pak",
       "//ui/resources:ui_test_pak",
     ]
-    deps = [ "//third_party/icu:icudata" ]
+    if (icu_use_data_file) {
+      sources += [ "$root_out_dir/icudtl.dat" ]
+      deps += [ "//third_party/icu:icudata" ]
+    }
 
     outputs = [ "{{bundle_resources_dir}}/{{source_file_part}}" ]
   }
